### PR TITLE
[Robot]GE:Review Donations Modal test case

### PR DESCRIPTION
**Pull request by skristem**
Added a new test that Creates an organization account with open opportunity (with payment record) via API. Go to SGE form select the donor as account and the account created. Verify review donations modal and select to update payment. Change date to today and payment amount to be less than the opp amount. Verify that the same payment record got updated with a new amount and date but the opportunity is still prospecting and the amount is not updated.
Also created a new API keyword that checks if GE is enabled via API first and if not enabled then enables via UI
AA WI-042803
File Created: review_donations_update_pmt_bge.robot
FIles Updated:
NPSP.py
NPSP.robot
change_labels_tmpbldr.robot
review_donations_create_opp_sge.robot
template_builder.robot
verify_mapped_fields_tmpbldr.robot
zedit_template.robot
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/resources/NPSP.py [hunk 1]
```diff
@@ -1588,6 +1588,8 @@ def click_special_button(self,title):
         """This keyword is similar to click button but uses set focus to button and javascript
```
Comment (rhakeshj) [inline robot/Cumulus/resources/NPSP.py:1588]: Can we update the method description with little more details?

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    ...                                  Donation Date=Today
+    Click Button                         Save & Enter New Gift
+    #verify donation date and amount values changed on table
+    Verify Gift Count                    1
+    Verify Table Field Values            Batch Gifts
+    ...                                  Donor Name=${ACCOUNT}[Name]
+    ...                                  Donation Amount=$499.50
+    ...                                  Donation Name=${PAYMENT}[Name]
+    ...                                  Donation Date=${UI_DATE}
+    Scroll Page To Location              0      0
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot:77]: is this needed ? can it be part of any of the keywords?

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    ...                  StageName=Prospecting    
+    ...                  Amount=500    
+    ...                  CloseDate=${FUT_DATE}    
+    ...                  npe01__Do_Not_Automatically_Create_Payment__c=false    
+    ...                  Name=${ACCOUNT}[Name] Donation
+    Set suite variable   &{OPPORTUNITY}
+    ${ui_date} =         Get Current Date                   result_format=%b %-d, %Y
+    Set suite variable   ${UI_DATE}
+    &{payment} =         API Query Record         npe01__OppPayment__c      npe01__Opportunity__c=${OPPORTUNITY}[Id]
+    Set suite variable   &{PAYMENT}
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot:35]: Did you investigate if we can make this part of the keyword Setupdata?

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+*** Keywords ***
+Setup Test Data
+    [Documentation]      Creates the organiation account,opportunity and queries the payment record required for the test
+    ...                  along with getting dates and namespace required for test.
+    &{account} =         API Create Organization Account    Name=${faker.company()}
+    Set suite variable   &{ACCOUNT}
+    ${fut_date} =            Get Current Date         result_format=%Y-%m-%d    increment=2 days
+    Set suite variable   ${FUT_DATE}
+    ${cur_date} =            Get Current Date         result_format=%Y-%m-%d
```
Comment (rjanjanam) [inline robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot:23]: Can you make the variable name to Uppercase during assignment too, to be consistent
     &{ACCOUNT} =         API Create Organization Account    Name=${faker.company()}
    Set suite variable   &{ACCOUNT}

File: robot/Cumulus/resources/NPSP.py [hunk 5]
```diff
@@ -1588,6 +1588,8 @@ def click_special_button(self,title):
...
         In the cases where javascript is being triggered on moving away from field, 
         click button doesn't seem to work in headless mode"""
+        actions = ActionChains(self.selenium.driver)
```
Comment (rjanjanam) [inline robot/Cumulus/resources/NPSP.py:1591]: Good use of ActionChains

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot (hunk 3 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot:35]: I tried doing this and if I use this it is taking 18 seconds to setup data verses 2 secs so using the straightforward method

File: robot/Cumulus/resources/NPSP.py (hunk 1 above)
Comment (skristem) [reply robot/Cumulus/resources/NPSP.py:1588]: done, please take a look again

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot (hunk 2 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot:77]: I already have a scroll element into view in the next keyword but somehow in this particular test case it is not sure to top enough for the button to be visible. Only this option worked in this case. I cannot add this to the following keyword "Click Gift Entry Button" as this keyword is used for various buttons and the buttons could be anywhere on the page

File: robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot (hunk 4 above)
Comment (skristem) [reply robot/Cumulus/tests/browser/gift_entry/review_donations_update_pmt_bge.robot]: done

**Pull request merged**